packages/plugin-ext/src/plugin/tree/tree-views.ts

```diff
@@ -312,20 +315,15 @@ class TreeViewExtImpl<T> implements Disposable {
             // parent is inconsistent
             return undefined;
         }
-        const idLabel = this.getTreeItemIdLabel(treeItem);
-        let possibleIndex = children.length;
-        // find the right element id by searching all possible id names in the cache
-        while (possibleIndex-- > 0) {
-            const candidateId = this.buildTreeItemId(parentId, possibleIndex, idLabel);
-            if (this.nodes.has(candidateId)) {
-                return chain.concat(candidateId);
-            }
+        const candidateId = this.buildTreeItemId(parentId, treeItem);
+        if (this.nodes.has(candidateId)) {
+            return chain.concat(candidateId);
```

Comment:
Does this new code mean we'll never hit the cache for tree items for which we compute an ID?

Before the logic seemed to assume we could find items from some `possibleIndex`, but now there's a case where `buildTreeItemId` generates a new unique ID which means that those items will never be found here.

Reply:
Only if the tree item does not have an id set. This is correct, in my opinion, because if it doesn't have an id, there's no way to tell it's the same tree item, so any state we might have for the tree item at index n might not apply to the new item at index n.

Reply:
Also, the documentation of `TreeItem` says:

>If not provided, an id is generated using the tree item's label. Note that when labels change, ids will change and that selection and expansion state cannot be kept stable anymore.

If you look at the way id's are constructed, the id's will be the same for items that have the same label.